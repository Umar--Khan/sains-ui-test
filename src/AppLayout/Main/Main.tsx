import { PropsWithChildren, ReactNode } from 'react';
import styled from 'styled-components';

const Container = styled.div`
  min-height: 50vh;
  max-width: 90rem;
  margin: 4rem auto;
  padding: 0 1rem;
  width: 100%;
  @media (min-width: 720px) {
    padding: 0;
    width: 70%;
  }
  @media (min-width: 720px) and (max-width: 1200px) {
    padding: 0;
    width: 75%;
  }
`;

const Main = ({ children }: PropsWithChildren<ReactNode>) => (
  <main>
    <Container>{children}</Container>
  </main>
);

export default Main;
