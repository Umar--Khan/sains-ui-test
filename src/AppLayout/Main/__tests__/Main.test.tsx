import renderer from 'react-test-renderer';

import Main from '../Main';

describe('<Main />', () => {
  it('Should render correctly', () => {
    const component = renderer.create(<Main>Main</Main>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
