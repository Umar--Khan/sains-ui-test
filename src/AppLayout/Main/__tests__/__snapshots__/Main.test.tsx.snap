// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render correctly 1`] = `
<main>
  <div
    className="sc-bdfBwQ kGjSLo"
  >
    Main
  </div>
</main>
`;
